crates/uv/src/commands/project/ephemeral.rs

```diff
@@ -0,0 +1,125 @@
+use tracing::debug;
+
+use cache_key::digest;
+use pypi_types::Requirement;
+use uv_cache::{Cache, CacheBucket};
+use uv_client::Connectivity;
+use uv_configuration::{Concurrency, PreviewMode};
+use uv_fs::{LockedFile, Simplified};
+use uv_python::{Interpreter, PythonEnvironment};
+use uv_requirements::RequirementsSpecification;
+use uv_resolver::Lock;
+
+use crate::commands::project::{resolve_environment, sync_environment};
+use crate::commands::SharedState;
+use crate::printer::Printer;
+use crate::settings::ResolverInstallerSettings;
+
+/// An ephemeral [`PythonEnvironment`] stored in the cache.
+#[derive(Debug)]
+pub(crate) struct EphemeralEnvironment(PythonEnvironment);
+
+impl From<EphemeralEnvironment> for PythonEnvironment {
+    fn from(ephemeral: EphemeralEnvironment) -> Self {
+        ephemeral.0
+    }
+}
+
+impl EphemeralEnvironment {
+    /// Get or create an [`EphemeralEnvironment`] based on a given set of requirements and a base
+    /// interpreter.
+    pub(crate) async fn get_or_create(
+        requirements: Vec<Requirement>,
+        interpreter: Interpreter,
+        settings: &ResolverInstallerSettings,
+        state: &SharedState,
+        preview: PreviewMode,
+        connectivity: Connectivity,
+        concurrency: Concurrency,
+        native_tls: bool,
```

Comment:
Once we have #4729 we should probably pass around a `BaseClientBuilder` instead of these individual options